symengine/symbol.h

```diff
@@ -46,12 +46,53 @@ class Symbol : public Basic
     }
 };
 
+class Dummy : public Symbol
+{
+private:
+    //! Dummy count
+    static size_t count_;
```

Comment:
`count_` is common to all the objects of class `Dummy`. You'll also need a `dummy_index` per object.